<h1 style="text-align:center; color:blue";>Cardiology Data Exploration with Multiple Linear Regression </h1>


In [1]:
# importing packages
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [2]:
#Loading the cardiology data into a Pandas dataframe
df = pd.read_csv("CardiologyNumerical.csv")

In [3]:
#Looking at the first 5 rows of the dataset

df.head()

angina  slope  thal  class
0       1      2     7      0
1       0      1     3      1
2       1      2     3      1
3       0      2     7      0
4       1      3     7      0

In [4]:
#Setting angina, slope, and thal values as the features and class as the output
X=df[['angina','slope','thal']]
y=df['class']

#Printing the shapes of the features and the output
print(X.shape)
print(y.shape)


(303, 3)
(303,)


In [5]:
#Importing scikit-learn's linear model and splitting the dataset into training (75%) and testing (25%) sets

from sklearn import linear_model
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.25,random_state=0)

print('X_train.shape: {}'.format(X_train.shape))
print('y_train.shape: {}'.format(y_train.shape))
print('X_test.shape: {}'.format(X_test.shape))
print('y_test.shape: {}'.format(y_test.shape))

X_train.shape: (227, 3)
y_train.shape: (227,)
X_test.shape: (76, 3)
y_test.shape: (76,)


In [6]:
#Fitting the data and printing each coefficient

regr = linear_model.LinearRegression()
regr.fit(X_train,y_train)
print(regr.coef_)

[-0.28367422 -0.13070114 -0.10220612]


In [14]:
#Investigating the regression score (based off of the testing data), and printing the intercept for the model's equation
print(regr.score(X_test,y_test))
print(regr.intercept_)

0.376548101768
1.32766970906


<h2>Creating the Regression Equation (the Model)</h2>

Class = (-0.28367422 x Angina) + (-0.13070114 x Slope) + (-0.10220612 x Thal) + 1.32766971

The model currently has an accuracy rating of 37.65%.

In [13]:
#Creating new experimental
X_new1=np.array([0,2,3])
X_new2=np.array([1,3,6])
X_new3=np.array([0,1,3])


#Reshaping the experimental data
X_new1=np.reshape(X_new1, (1,-1))
X_new2=np.reshape(X_new2, (1,-1))
X_new3=np.reshape(X_new3, (1,-1))

#Printing the reshaped data
print('X_new1 shape: {}'.format(X_new1.shape))
print('X_new2 shape: {}'.format(X_new2.shape))
print('X_new3 shape: {}'.format(X_new3.shape))
print(y.shape)




X_new1 shape: (1, 3)
X_new2 shape: (1, 3)
X_new3 shape: (1, 3)
(303,)


In [9]:
#Running the reshaped experimental data through the model to see what the predictions are

prediction1 = regr.predict(X_new1)
print('Prediction is: {}'.format(prediction1))

prediction2 = regr.predict(X_new2)
print('Prediction is: {}'.format(prediction2))

prediction3 = regr.predict(X_new3)
print('Prediction is: {}'.format(prediction3))

Prediction is: [ 0.75964907]
Prediction is: [ 0.03865535]
Prediction is: [ 0.89035021]


While not dead on accurate (correct answers 1,0,1), if the answers are rounded up/down then the model was correct in all three of its predictions.